### PR TITLE
bump: update Golang to v1.22.3

### PR DESCRIPTION
**Pull request by qweeah**
**What this PR does / why we need it**:
This PR bumps all used Golang to v1.22.3

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module oras.land/oras
 
-go 1.22.0
```
Comment (shizhMSFT) [inline go.mod:3]: We don't need to update the minimum go version in go mod, do we?

Comment (qweeah) [reply go.mod:3]: You mean patch version? Tried `go mod init test` and it gives me patch version go.mod
```go.mod
module test

go 1.22.1
```

Comment (qweeah) [reply go.mod:3]: So yes we should update go patch version.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**